pkg/controller/job/controller.go

```diff
@@ -403,10 +398,6 @@ func (jm *JobManager) manageJob(activePods []*api.Pod, successful, unsuccessful
 	} else if active < parallelism {
 		// how many executions are left to run
 		diff := *job.Spec.Completions - successful
-		// for RestartPolicyNever we need to count unsuccessful pods as well
```

Comment:
Why this was removed? Without it, the comment in [types.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/experimental/types.go#L437-L439) does not make sense, since we'll always update the unsuccessful value. Besides, this PR also change the idea of not restarting failed pods when RestartPolicyNever is set. Currently the RestartPolicy does not change behavior of job controller, is that indented? May I know the reasoning?